homeassistant/components/prometheus/__init__.py

```diff
@@ -315,28 +317,53 @@ def _handle_lock(self, state):
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
-    def _handle_climate(self, state):
-        temp = state.attributes.get(ATTR_TEMPERATURE)
+    def _handle_climate_temp(
+        self, state, attr, metric_name, metric_description, extra_labels=None
+    ):
+        if extra_labels is None:
+            extra_labels = {}
+
+        temp = state.attributes.get(attr)
         if temp:
             if self._climate_units == TEMP_FAHRENHEIT:
                 temp = fahrenheit_to_celsius(temp)
             metric = self._metric(
-                "climate_target_temperature_celsius",
+                metric_name,
                 self.prometheus_cli.Gauge,
-                "Target temperature in degrees Celsius",
+                metric_description,
+                extra_labels.keys(),
             )
-            metric.labels(**self._labels(state)).set(temp)
+            all_labels = {**extra_labels, **self._labels(state)}
+            metric.labels(**all_labels).set(temp)
 
-        current_temp = state.attributes.get(ATTR_CURRENT_TEMPERATURE)
-        if current_temp:
-            if self._climate_units == TEMP_FAHRENHEIT:
-                current_temp = fahrenheit_to_celsius(current_temp)
-            metric = self._metric(
-                "climate_current_temperature_celsius",
-                self.prometheus_cli.Gauge,
-                "Current temperature in degrees Celsius",
-            )
-            metric.labels(**self._labels(state)).set(current_temp)
+    def _handle_climate(self, state):
```

Comment:
The way this is structured now makes it seem like it's possible to set different descriptions and label keys for each of the three uses of `climate_target_temperature_celsius`, while in practice the metric will be created when it's first referenced.

This might be surprising to someone who is going to be looking at this code or changing it in the future.

Do you think there is a way to make the description and the label names defined once for each metric?

Reply:
I'll just switch back to using three different metrics, that will avoid the issue you're describing